packages/medusa/src/services/product.ts

```diff
@@ -775,6 +827,29 @@ class ProductService extends TransactionBaseService<ProductService> {
       q,
     }
   }
+
+  private async handleCreateOrUpdateWithSalesChannelErrorIfNecessary(
+    salesChannelIds: string[]
+  ): Promise<never | void> {
+    if (salesChannelIds?.length) {
```

Comment:
nit: `salesChannelIds` is not optional
also could we do this check outside so that the method is only responsible for throwing and then we can rename to `throwCreateOrUpdateWithSalesChannelError`

Reply:
It is not optional but it can be empty 😉 but indeed i can move the statement outside

Reply:
I was suggesting to remove "?"  initially since from the method definition `salesChannelIds` will always be an array

Reply:
Ahah yes true 😂